fetch/ledger/api/contracts.py

```diff
@@ -0,0 +1,95 @@
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+from fetch.ledger.api import ApiEndpoint, ApiError
+from fetch.ledger.serialisation.objects.transaction_api import *
+from fetch.ledger.crypto.signing import Signing
+
+import base64
+import hashlib
+import binascii
+
+class SmartContractManager(ApiEndpoint):
+    API_PREFIX = 'fetch.smart_contract_manager'
+
+    def add_contract(self, contract_source):
+        """
+        Add a smart contract to the ledger
+        """
+
+        contract_source_b64 = base64.b64encode(contract_source.encode()) # Contract source as base64
+        contract_hash_b64   = base64.b64encode(hashlib.sha256(contract_source_b64).digest()) # base64 of sha256 of the contract_source
+
+        contract_source_b64 = contract_source_b64.decode()
+        contract_hash_b64   = contract_hash_b64.decode()
+
+        # Generate random pub-private key pair for this submission
```

Comment:
all of these examples generate a new Pub/Pri pair - instead they should take them as input.

Btw, a bit annoying there aren't any intuitive 'example scripts' that drive the fetch api with a single ./ 

There needs to be some consistency about what types are going to be passed around in this api. I believe tokenapi takes bytes into one arg for a private key and a base64 string for another PK. adding in there is no type checking it could be more intuitive